py/tests/test_fetcher.py

```diff
@@ -0,0 +1,207 @@
+import copy
+import os
+import unittest
+from io import StringIO
+from unittest.mock import Mock, call, patch
+
+from typing_extensions import Mapping, Sequence, cast
+
+from tests.test_data import datadog_response
+from usageaccountant import accumulator
+from usageaccountant import datadog_fetcher as ddf
+
+
+def read_data_file(filename: str) -> str:
+    tests_dir = os.path.dirname(os.path.abspath(__file__))
+    query_file_path = os.path.join(tests_dir, "test_data", filename)
+    with open(query_file_path, "r") as file:
+        return file.read()
```

Comment:
Please use temporary directories for tests, it will be easier, so you do not depend on where the test file is.
The config file is very small so you can inline it inside the test.

Example: https://github.com/getsentry/ops/blob/master/k8s/cli/assistant/test_prdocs.py#L10-L38

1.  `with tempfile.TemporaryDirectory() as temp_dir:` creates a temp dir during your test
2.  `with open(full_path1 / "topic1.yaml", "w") as file:` opens the file and then you write the content.

Reply:
I abandoned files altogether and put everything inline.